src/backend/distributed/executor/local_executor.c

```diff
@@ -206,7 +207,10 @@ ExecuteLocalTaskList(CitusScanState *scanState, List *taskList)
 			localPlan = planner(shardQuery, cursorOptions, paramListInfo);
 		}
 
-		LogLocalCommand(task);
+		if (ShouldLogLocalCommand())
```

Comment:
This works.

Reply:
Could use a comment, e.g.
```c
/*
 * We avoid calling TaskQueryString unless the user wants the query to be logged,
 * since computing the deparsed query string is not needed when using a cached local 
 * plan.
 */
```

Reply:
@marcocitus  is this good to merge?